src/type-checker.cc

```diff
@@ -33,7 +33,8 @@ std::string TypesToString(const TypeVector& types,
     Type ty = types[i];
     // NOTE: Reference (and GetName) is also used by (e.g.) objdump, which does
     // not apply validation. do this here so as to not break that.
-    if (ty == Type::Reference && ty.GetReferenceIndex() == kInvalidIndex) {
+    if (ty.code() == Type::Reference &&
```

Comment:
I would prefer to keep these this this way, because `ty == Type::Reference` is misleading equality for me, since the index is not checked.